### ISSUE TITLE
False positive implicit yield violation

### ISSUE THREAD

**Issue by AlwxSin** [wemake-services/wemake-python-styleguide#1057]
# Bug report

## What's wrong
With common function to split iterable by chunks WPS raises `ImplicitYieldFromViolation`

```
def chunks(value, chunk_size):
    """Yield successive n-sized chunks from l."""
    for index in range(0, len(value), chunk_size):
        yield value[index:index + chunk_size]
```

## How is that should be
`ImplicitYieldFromViolation` should be raised only if no other operations with iterable item.

**Comment by sobolevn** [comment]
What about 

```python
yield from (
    value[index:index + chunk_size]
    for index in range(0, len(value), chunk_size)
)
```

?

**Comment by AlwxSin** [comment]
Less readable in my taste. But if no other choices I will use it. Thanks.

**Comment by sobolevn** [comment]
Thanks for raising this, I guess we need to update the violation docs about this case.

**Comment by AlwxSin** [comment]
@sobolevn there is still a bug. 
`Python does not support 'yield from' inside async functions`

**Comment by sobolevn** [comment]
Yes, that's true.

**Comment by sobolevn** [comment]
This rule should be ignored for `async` functions.

Can you please add a check that `if isinstance(get_context(node), ast.AsyncFunctionDef): return`?

**Comment by AlwxSin** [comment]
Of course, I'll do it.
